WordPress/src/main/java/org/wordpress/android/modules/ApplicationModule.java

```diff
@@ -1,15 +1,40 @@
 package org.wordpress.android.modules;
 
 import android.app.Application;
+import android.arch.lifecycle.ViewModel;
+import android.arch.lifecycle.ViewModelProvider;
 import android.content.Context;
+import android.support.v4.util.ArrayMap;
+
+import org.wordpress.android.viewmodel.PluginBrowserViewModel;
+
+import javax.inject.Provider;
+import javax.inject.Singleton;
 
 import dagger.Binds;
 import dagger.Module;
+import dagger.Provides;
 
-@Module
+@Module(subcomponents = ViewModelSubComponent.class)
 public abstract class ApplicationModule {
     // Expose Application as an injectable context
     @Binds
     abstract Context bindContext(Application application);
+
+    @Provides
+    @Singleton
+    public static ViewModelProvider.Factory provideViewModelFactory(ViewModelSubComponent.Builder subComponentBuilder) {
+        ArrayMap<Class<? extends ViewModel>, Provider<ViewModel>> viewModelMap = new ArrayMap<>();
+        final ViewModelSubComponent subComp = subComponentBuilder.build();
+        viewModelMap.put(PluginBrowserViewModel.class, new Provider<ViewModel>() {
+            @Override
+            public ViewModel get() {
+                return subComp.pluginBrowserVM();
+            }
+        });
+        return new ViewModelFactory(viewModelMap);
```

Comment:
Question: I am just curious, is there any documentation/article I can read about this pattern? I've played with Google's samples and they take a different approach. I think I only saw one subcomponent approach in the articles I read, but I don't really have a good grasp on when one is applicable over the other or what the advantages/disadvantages are.

Reply:
I wanted to encapsulate ViewModel related staff into a separate component as explained [here](https://google.github.io/dagger/subcomponents.html). I'm not sure whether I got an inspiration somewhere when I used this pattern for the first time or if I just felt (thought) it makes sense.

Reply:
@oguzkocer Can you please send me an example of Google's solution? I've tried to find it in the samples, but without any luck.

Reply:
Here it is: https://github.com/googlesamples/android-architecture-components/tree/master/GithubBrowserSample